modular_citadel/code/modules/festive/vectorcraft.dm

```diff
@@ -0,0 +1,585 @@
+		message_admins("Pix:[pixel_x],[pixel_y] TileLoc:[tile_loc["x"]], [tile_loc["y"]]. [round(tile_loc["x"])], [round(tile_loc["y"])]")
+	//no tile movement
+
+	if(x_move == 0 && y_move == 0)
+		return FALSE
+
+	//var/direction = calc_step_angle(x_move, y_move)
+	//if(direction) //If the movement is greater than 2
+	//	step(src, direction)
+	//	after_move(direction)
+
+
+
+
+	return TRUE
+
+//////////////////////////////////////////////////////////////
+//					Check procs						    	//
+//////////////////////////////////////////////////////////////
+
+//check the cooldown on the boost
+/obj/vehicle/sealed/vectorcraft/proc/check_boost()
+	if(enginesound_delay < world.time)
+		enginesound_delay = 0
+	if(!boost_cooldown)
+		return
+	if(boost_cooldown < world.time)
+		boost_cooldown = 0
+		playsound(src.loc,'sound/vehicles/boost_ready.ogg', 65, 0)
+	return
+
+//Make sure the clutch is on while changing gears!!
+/obj/vehicle/sealed/vectorcraft/proc/check_gears()
+	if(!driver)
+		for(var/i in contents)
+			if(iscarbon(i))
+				var/mob/living/carbon/C = i
+				driver = C
+				to_chat(driver, "<span class='notice'><b>You shuffle across to the driver's seat of the [src]</b></span>")
+				start_engine()
+				break
+		if(!driver)
+			stop_engine()
+			return
+	if(!gear)
+		gear = driver.a_intent
+	if(gear == "auto")
+		return
+	//USE THE CCLUUUTCHHH
+	if(gear != driver.a_intent && !driver.combatmode)
+		//playsound
+		to_chat(driver, "<span class='warning'><b>The gearbox gives out a horrific sound!</b></span>")
+		playsound(src.loc,'sound/vehicles/clutch_fail.ogg', 70, 0)
+		apply_damage(5)
+		acceleration = acceleration/2
+	else if(gear != driver.a_intent && driver.combatmode)
+		playsound(src.loc,'sound/vehicles/clutch_win.ogg', 100, 0)
+	gear = driver.a_intent
+
+//Bounce the car off a wall
+/obj/vehicle/sealed/vectorcraft/proc/bounce()
+	vector["x"] = -vector["x"]/2
+	vector["y"] = -vector["y"]/2
+	acceleration /= 2
+
+/obj/vehicle/sealed/vectorcraft/proc/ricochet(x_move, y_move)
+	var/speed = calc_speed()
+	apply_damage(speed/10)
+	bounce()
+
+//////////////////////////////////////////////////////////////
+//					Damage procs							//
+//////////////////////////////////////////////////////////////
+//Repairing
+/obj/vehicle/sealed/vectorcraft/attackby(obj/item/O, mob/user, params)
+	.=..()
+	if(istype(O, /obj/item/weldingtool) && user.a_intent != INTENT_HARM)
+		if(obj_integrity < max_integrity)
+			if(!O.tool_start_check(user, amount=0))
+				return
+
+			user.visible_message("[user] begins repairing [src].", \
+				"<span class='notice'>You begin repairing [src]...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+
+			if(O.use_tool(src, user, 40, volume=50))
+				to_chat(user, "<span class='notice'>You repair [src].</span>")
+				apply_damage(-max_integrity)
+		else
+			to_chat(user, "<span class='notice'>[src] does not need repairs.</span>")
+
+//Heals/damages the car
+/obj/vehicle/sealed/vectorcraft/proc/apply_damage(damage)
```

Comment:
I.. really won't bother you about "unsnowflakeying" this function right now, but keep in mind about the existence of `take_damage()` and `obj_break()` for later.